### ISSUE TITLE
Issue with associate when using a LM for retrival

### ISSUE THREAD

**Issue by pratkpranav** [ThirdAILabs/Demos#94]
This is one of the issues that one of the participants in the hackathon faced, he associated `sbf -> Sam Bankman Fried`. However, when he is trying the query `who is sbf?` with OpenAI APIs which context consists of text with only `Sam Bankman Fried` It was saying `I cannot answer`.  Since OpenAI doesn't have the notion that sbf could be referred to as Sam Bankman Fried.

**Comment by pratkpranav** [comment]
@tharunkr24 @dtr3

**Comment by dtr3** [comment]
oh! Since we log all associate data in neural db, maybe we can add that to the gen ai prompt?

"Assume sbf means Sam Bankman Fried"
... etc for all associate data

**Comment by pratkpranav** [comment]
Yes, something similar could work, though it can explode quickly with more associations, we can get in the case where we are passing more associations than the actual context. We could change the outputs/context itself(replacing the text with their associated text) based on the query and associations we have to give relevant results.

Though I understand it's trickier to solve that way what if the user is referring to original text? Any inputs @tharunkr24?
